mgmtfn/dockplugin/netDriver.go

```diff
@@ -523,3 +638,174 @@ func GetDockerNetworkName(nwID string) (string, string, string, error) {
 
 	return tenantName, netName, serviceName, nil
 }
+
+// FindGroupFromTag finds the group that has matching tag
+func FindGroupFromTag(epgTag string) (*mastercfg.EndpointGroupState, error) {
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return nil, err
+	}
+
+	epgCfg := &mastercfg.EndpointGroupState{}
+	epgCfg.StateDriver = stateDriver
+
+	epgList, err := epgCfg.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+
+	var epg *mastercfg.EndpointGroupState
+	found := false
+	for _, epgP := range epgList {
+		epg = epgP.(*mastercfg.EndpointGroupState)
+		if epg.GroupTag == epgTag {
+			found = true
+			break
+		}
+	}
+	if !found {
+		err = errors.New("Couldn't find group matching the tag")
+		return nil, err
+	}
+	return epg, nil
+}
+
+// FindNetworkFromTag finds the network that has matching tag
+func FindNetworkFromTag(nwTag string) (*mastercfg.CfgNetworkState, error) {
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return nil, err
+	}
+	nwCfg := &mastercfg.CfgNetworkState{}
+	nwCfg.StateDriver = stateDriver
+	nwList, err := nwCfg.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	var nw *mastercfg.CfgNetworkState
+	found := false
+	for _, nwP := range nwList {
+		nw = nwP.(*mastercfg.CfgNetworkState)
+		if nw.NetworkTag == nwTag {
+			found = true
+			break
+		}
+	}
+	if !found {
+		err = errors.New("Couldn't find network matching the tag")
+		return nil, err
+	}
+	return nw, nil
+}
+
+// createNetworkHelper creates the association between docker network and contiv network
+//  if tag is given map docker net to epg or network, else create a contiv network
+func createNetworkHelper(networkID string, tag string, IPv4Data, IPv6Data []driverapi.IPAMData) error {
+	var tenantName, networkName, serviceName string
+	var err error
+	if tag != "" {
+		// we need to map docker network to policy group or network using the tag
+		log.Infof("Received tag %s", tag)
+		var nw *mastercfg.CfgNetworkState
+		epg, err := FindGroupFromTag(tag)
+		if err != nil {
+			nw, err = FindNetworkFromTag(tag)
+			if err != nil {
+				return errors.New("failed to lookup tag")
```

Comment:
Not sure if we should propagate the internal errors to docker API response. We have log.Errorf in contiv for more debugging